contributing/principles.md

```diff
@@ -0,0 +1,90 @@
+[&larr; Table of Contents](../CONTRIBUTING.md)
+
+# Guiding Principles
+💡 The _what_ and _why_ of Form Design System software.
+
+## Purpose of the design system
+At CB Insights, our design system has a number of goals that aim to help us build a
+world-class UI:
+
+- _Record design and implementation decisions for reusable UI_
+- _Foster consistency in visual design and behavior of UI_
+- _Enable front end engineers to pool their knowledge into a single point of truth for UI components_
+- _Help feature teams spend less time on UI details and more time on solving business problems_
+
+## What belongs in FDS?
+All styles and components in FDS should be application-agnostic. We should be able to use
+the FDS library within `cbi-site`, admins, and any other front end CB Insights may build.
+
+### ✅ Belongs in FDS
+- Low specificity, global utility classes
+- React layout utilities and other low-level React helpers
+- Presentational components that are not tied to any specific feature or application
+
+### 🚫 Does not belong in FDS
+- Components that rely on Redux state
+- Components that are tied to a specific application or feature
+- Components that are not "small and sharp" (see below)
+
+## Alignment with design
+When it comes to standard UI components, Engineering and Design should speak the same
+language. In practice, this means:
+
+- Publicly exposed components in FDS should be named based on how design refers to the
+component in their sketch library
```

Comment:
I 100% agree but this being the case, we'll want to start doubling down to make sure design doesn't reflect business use cases (There's been at least one component so far which we've discussed that calls out specific business cases. )

Reply:
This sounds like more of a thing for us to keep in mind as we move forward. Design so far has been basing common components off of feature designs which definitely include business.

Is there anything we should add to these docs regarding that though?

Reply:
hmmm... yeah this is tough. I think this comes back to a lot of this: https://github.com/cbinsights/form-design-system/pull/88#pullrequestreview-277219170

Essentially, do we want to spell out that developers should push back against business logic being inserted into components, or push more on the design side so that there isn't.

Me and Asaf have been having some really good discussions as of late that may inform some of this as well (we should talk about this in our 1:1 probably). I think we can easily just say that we'll come back to this.

Reply:
I think it goes back to the point I had at the very very top: https://github.com/cbinsights/form-design-system/pull/88#pullrequestreview-277219170

I think this is something worth covering in our 1:1 though, for us to think about and we can always come back if we so choose.

Reply:
Sounds to me like this is a wider concern than what contributing guidelines for FDS can cover. Let's discuss how best to encourage pushing back from feature teams